### DISCUSSION TITLE
构建需求整理

### DISCUSSION THREAD

**Discussion by zanyzhao** [Tencent/bk-ci#4413] [General]
## 机器状态
- 显示每台构建机的  ip地址，名字，每个盘的空间使用率，内存，cpu，是否在线 等信息。
- 另外想要知道各机器的负载情况，比如，对于某个机器A，希望可以统计：
  - 一天内有多少时间是有构建任务正在运行的，有多少时间是空闲的。
  - 同一时刻内，并行运行的构建任务有几个

## 流水线整理
根据项目，导出该项目下面的所有流水线（以json配置或者其他形式的配置来呈现）
目的：方便项目组整理，比如我们可以根据这些配置轻松的统计出  主干版本 总共占用了几台构建机，分别是哪些流水线。

## 流水线本身
选定某流水线，可以看到该流水线的构建成功/失败概率，每次成功构建的时间消耗，便于我们发现某些构建效率问题，比如这样
![image](https://user-images.githubusercontent.com/1629409/121677596-ab4b6300-cae8-11eb-80fc-17107d117fb4.png)

## 流水线机器与机器之前的通信
目前，构建机之间互相拷贝文件只能人肉操作，希望可以出一个蓝盾插件，指定某构建机，某目录，然后可以把文件传输到另一台构建机上面去

## 构建配置相关
对于同一个构建任务：
- 填了一堆参数，启动构建，构建成功，构建号是#314
- 又填了一堆参数，启动构建，构建成功，构建号是#315
但是#314号构建和#315号构建这两次构建任务，有哪些参数不同，目前只能通过人肉一个一个去对比。希望可以统一把 #314的所有配置参数导出，把#315构建的所有配置参数导出，方便我们对比查看

**Comment by zanyzhao** [comment]
> * 显示每台构建机的  ip地址，名字，每个盘的空间使用率，内存，cpu，是否在线 等信息。

是否在线的信息已经在【环境管理-节点详情页】具备了
![image](https://user-images.githubusercontent.com/1629409/122148592-506d8f00-ce8d-11eb-8145-710ea9330abd.png)

IP地址、名字、磁盘使用率、CPU、内存等信息也在节点详情页有了
![image](https://user-images.githubusercontent.com/1629409/122148772-9c203880-ce8d-11eb-806a-9f1fb60ba33a.png)

**Reply by jackson002** [reply]
嗯，目前的的确有信息
但是在展示方面可能不太能满足
想要的是能够一个页面就直接能看到所有的机器信息

**Reply by zanyzhao** [reply]
目前的列表页数据项都是最基础的必要信息，已经不少了，再加些就没法展示了。

其实你人肉分析这些数据的目的也无非是希望让构建机的分布更合理，有没有可能把你现在分配构建机的规则抽象成规则，我们在调度时引用你的规则，可以从根本上解决这问题呢？

**Comment by zanyzhao** [comment]
> 流水线整理

已录入issue #4429 ，请关注

**Comment by zanyzhao** [comment]
> 选定某流水线，可以看到该流水线的构建成功/失败概率，每次成功构建的时间消耗，便于我们发现某些构建效率问题，比如这样

合理的需求，我们尽快安排 #4434，争取7月份搞定

**Comment by zanyzhao** [comment]
> 目前，构建机之间互相拷贝文件只能人肉操作，希望可以出一个蓝盾插件，指定某构建机，某目录，然后可以把文件传输到另一台构建机上面去

目前内置了“归档构件”和“拉取已归档构件”，可以实现类似效果

**Comment by zanyzhao** [comment]
> ## 构建配置相关
> 对于同一个构建任务：
> 
> * 填了一堆参数，启动构建，构建成功，构建号是#314
> * 又填了一堆参数，启动构建，构建成功，构建号是#315
>   但是#314号构建和#315号构建这两次构建任务，有哪些参数不同，目前只能通过人肉一个一个去对比。希望可以统一把 #314的所有配置参数导出，把#315构建的所有配置参数导出，方便我们对比查看

已经有类似issue #4016 ，也是排入7月份
